core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -558,6 +560,57 @@ private static Optional<TServerInstance> checkServer(ClientContext context, Stri
         .map(address -> new TServerInstance(address, stat.getEphemeralOwner()));
   }
 
+  public static void validate(TabletMetadata tm) {
+    if (tm == null) {
+      throw new IllegalStateException("TabletMetadata cannot be null");
+    }
+
+    Text prevEndRowText = tm.getPrevEndRow();
+    Text endRowText = tm.getEndRow();
+
+    // Allow validation even if prevEndRow is missing, as long as endRowText exists
+    Key prevEndRowKey = (prevEndRowText != null) ? new Key(prevEndRowText) : null;
+    Key endRowKey = (endRowText != null) ? new Key(endRowText) : null;
+
+    Collection<StoredTabletFile> files = tm.getFiles();
+    if (files == null || files.isEmpty()) {
+      return;
+    }
```

Comment:
I think you can move this up before you get the prevEndRow and endRow. If there are no files to validate, then just exit early.